### PR TITLE
feat: folder inheritance  scripts

### PR DESCRIPTION
**Pull request by jackkav**
todo
- [x] add tab and edit pane
- [x] wire up
- [x] test

closes INS-3860

challenges
- how to wrap each level of script in such a way they don't affect the promises or the variables scopes of others

### CODE REVIEW

File: packages/insomnia/src/models/request-group.ts [hunk 1]
```diff
@@ -18,7 +18,7 @@ interface BaseRequestGroup {
   environmentPropertyOrder: Record<string, any> | null;
   metaSortKey: number;
   preRequestScript: string;
-  postRequestScript: string;
+  afterRequestScript: string;
```
Comment (ihexxa) [inline packages/insomnia/src/models/request-group.ts]: we might named it as `afterResponseScript` after merging.

File: packages/insomnia/src/ui/components/panes/request-group-pane.tsx [hunk 2]
```diff
@@ -119,6 +122,60 @@ export const RequestGroupPane: FC<{ settings: Settings }> = ({ }) => {
...
+            />
+          </ErrorBoundary>
+        </TabItem>
+        <TabItem
+          key="after-request-script"
+          data-testid="after-request-script-tab"
+          title={
+            <div className='flex items-center gap-2'>
+              After-request Script{' '}
+              {activeRequestGroup.preRequestScript && (
```
Comment (ihexxa) [inline packages/insomnia/src/ui/components/panes/request-group-pane.tsx]: It should be `activeRequestGroup.afterResponseScript` instead `preXX`.

File: packages/insomnia/src/network/network.ts [hunk 3]
```diff
@@ -132,9 +132,21 @@ export const tryToExecutePreRequestScript = async (
...
+    ]) as (Request | RequestGroup)[];
+
+    const folderScripts = requestGroups.reverse()
+      .filter(group => group?.preRequestScript)
+      .map((group, i) => `const fn${i} = async ()=>{
+          ${group.preRequestScript}
+        }
+        await fn${i}();
+    `);
+    const joinedScript = [...folderScripts].join('\n');
```
Comment (ihexxa) [inline packages/insomnia/src/network/network.ts]: we might also think about things like displaying progress or proper timeout value for joined scripts.

File: packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts [hunk 4]
```diff
@@ -451,18 +451,6 @@ test.describe('unhappy paths', async () => {
     });
 
     const testCases = [
-        {
```
Comment (ihexxa) [inline packages/insomnia-smoke-test/tests/smoke/pre-request-script-features.test.ts:454]: Ah we might bring back this case.

**Review by gatzjames — APPROVED** [review]

**Pull request merged**